### PR TITLE
Package update - Spring 2017

### PR DESCRIPTION
**Pull request by Maxyme**
An update of the CG recipes. Is it time to give noarch a chance?

### CODE REVIEW

File: azure-storage/meta.yaml [hunk 1]
```diff
@@ -18,8 +19,7 @@ requirements:
 
   run:
     - python
-    - azure-nspkg >=1.0.0
-    - azure-common >=1.1.4
```
Comment (Korijn) [inline azure-storage/meta.yaml]: are you sure nspkg is no longer required?

File: pydocstyle/meta.yaml [hunk 2]
```diff
@@ -1,14 +1,15 @@
 package:
   name: pydocstyle
-  version: 1.1.1
+  version: 2.0.0
```
Comment (Korijn) [inline pydocstyle/meta.yaml]: I expect this to introduce a lot of new linting errors, they added numpydoc support I think

File: sphinx-gallery/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package:
+  name: sphinx-gallery
```
Comment (Korijn) [inline sphinx-gallery/meta.yaml]: didn't we already have this one?

File: azure-storage/meta.yaml (hunk 1 above)
Comment (Maxyme) [inline azure-storage/meta.yaml]: yes - 
https://github.com/Azure/azure-storage-python/commit/38662e7958078929fb750faf05ad1c1ad6d6fc4e#diff-2eeaed663bd0d25b7e608891384b7298R73

File: pydocstyle/meta.yaml (hunk 2 above)
Comment (Maxyme) [inline pydocstyle/meta.yaml]: It's a dependency for flake8-docstring. 
here are the release notes. http://www.pydocstyle.org/en/latest/release_notes.html

File: sphinx-gallery/meta.yaml (hunk 3 above)
Comment (Maxyme) [inline sphinx-gallery/meta.yaml]: Surprisingly no, I added it.

**Pull request merged**